students/travis_nelson/lesson05/mailroom_part3.py

```diff
@@ -0,0 +1,163 @@
+            "Paul Allen": [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0],
+            "Brax Dingle": [2331, 32322.87, 5566.20, 3323.23, 76, 323, 3.87],
+            }
+
+
+def single_thank_you():
+    """Prints a letter to user-specified donor,
+    including recent and historical donation amounts"""
+    recipient = recipient_response()
+    donation = donation_response(recipient)
+    store_new_donation(recipient, donation)
+    total_donations = sum_donations(recipient)
+    print("Dearest {},\n"
+          "We are writing to formally thank you for your generous"
+          " donation of ${}.\n"
+          "To date, you have donated ${} to our honorable mission.\n"
+          "You are truly a valuable patron, and we thank you"
+          " for your service"
+          " (and money)".format(recipient, donation, total_donations))
+
+
+def recipient_response():
+    """Returns a recipient from the user...
+    If recipient was not previously in donor database, they are added"""
+    keep_asking = True
+    while keep_asking:
+        response = input("\nTo whom should we address this Thank You?\n"
+                         "(Type 'list' to view existing donors)\n\n").title()
+        try:
+            if response[0].isalpha():
+                if response.lower() == "list":
+                    print("\n")
+                    for i in donor_db:
+                        print("{}".format(i))
+                else:
+                    donor_exists = False
+                    for donor in donor_db:
+                        if donor == response:
+                            donor_exists = True
+                    if donor_exists:
+                        print("\nAn existing patron!\n")
+                        return response
+                    else:
+                        print("\nA new donor!\n")
+                        donor_db[response] = []
+                        return response
+            else:
+                raise TypeError
+        except TypeError:
+            print("\nType a name, plz\n")
+
+
+def donation_response(donor):
+    """Returns the donation amount from the user"""
+    keep_asking = True
+    while keep_asking:
+        donation = input("How much did they donate?\n")
+        try:
+            if float(donation):
+                print("\nNice!\n")
+                keep_asking = False
+                return donation
+        except ValueError:
+            print("\nJust a number is fine.\n")
+
+
+def store_new_donation(donor, donation):
+    """Stores the donation amount in the donor database"""
+    updated_donations = donor_db[donor]
+    updated_donations.append(float(donation))
+    donor_db[donor] = updated_donations
+
+
+def sum_donations(donor_name):
+    """Returns sum of all historical donor donations"""
+    return round(sum(donor_db[donor_name]), 2)
+
+
+def sort_key(donor):
+    """Returns the key to sort donations by"""
+    return round(sum(donor_db[donor]))
```

Comment:
very small nitpick: it's good to have consistency 
`def sum_donations(donor_name):` and `def sort_key(donor):` have the same arguments that are named differently.